packages/playwright-core/src/server/registry/index.ts

```diff
@@ -1025,6 +1025,17 @@ export class Registry {
     const lockfilePath = path.join(registryDirectory, '__dirlock');
     const linksDir = path.join(registryDirectory, '.links');
 
+    // proper-lockfile does retry EACCES errors as well hence we do a best-effort permission check here to not end up stalling.
+    try {
+      const lockfileTestPath = path.join(registryDirectory, '__dirlock-test-' + process.pid);
+      await fs.promises.writeFile(lockfileTestPath, '');
+      await fs.promises.unlink(lockfileTestPath);
+    } catch (e) {
+      if (e.code === 'EACCES' || e.code === 'EPERM') {
+        throw new Error(`Permission denied: ${registryDirectory}.\n` +
+          `Please check if you have write permissions to the directory.`);
+      }
+    }
```

Comment:
Unrelated to this PR, but I noticed that `uninstall` command is not protected by the same lock. Sounds like a bug?

Reply:
https://github.com/microsoft/playwright/pull/35422